include/core/physical_property_model.h

```diff
@@ -96,6 +97,7 @@ class PhysicalPropertyModel
     model_depends_on[temperature_p2]            = false;
     model_depends_on[pressure]                  = false;
     model_depends_on[phase_order_cahn_hilliard] = false;
+    model_depends_on[levelset]                  = false;
```

Comment:
The way you are making it is by modifying the physical property model. Which is fine. In this case, however, this would a fluid property model (ImmersedSolid) instead of requiring  an additional solid subsection for properties.

Reply:
So in the physical properties subsection, we would have fluids and solids, but instead of `constant` model it would be `immersed solid`?